File: NewsApp/app/src/main/java/com/example/android/topic41/data/network/JsonPlaceHolderApi.java
package com.example.android.topic41.data.network;

import com.example.android.topic41.domain.util.News;

import java.util.Map;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface JsonPlaceHolderApi {
    @GET("/v2/top-headlines")
    Single<News> getNews(@QueryMap Map<String, String> parameters);
}

File: NewsApp/app/src/main/java/com/example/android/topic41/data/network/JsonPlaceHolderApi.kt
package com.example.android.topic41.data.network

import com.example.android.topic41.domain.util.News
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface JsonPlaceHolderApi {
    @GET("/v2/top-headlines")
    fun getNews(@QueryMap parameters: Map<String, String>): Single<News>
}